README.md

```diff
@@ -145,7 +168,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+Command to run my code: ./answer.exe
```

Comment:
Ummm... what's this `.exe` of which you speak? :)

Nice code. I was, however, hoping to see the state transitions expressed in a table (hence the previous question). Something like:

~~~ cpp
#include <iostream>
using namespace std;
#include <map>

typedef int state;
typedef std::map<std::string, state> transitions;
typedef std::map<state, transitions> state_table;

const int END = 99;

state_table states = {
  { 0, {{ "the", 1 }}},
	{ 1, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 2, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 3, {{ "ate", 4},  {"ran",    4}}},
	{ 4, {{ "slowly", 5}, { "noisily", 5}, {"EOI", END}}},
	{ 5, {{ "EOI", END }}}
};


int getNextState(int currentState, string nextWord) {
	transitions t = states[currentState];
	if (t.count(nextWord))
		return t[nextWord];
	else
		return -1;
}
~~~

# 10